### PR TITLE
Add generic quad airframes for 250, 450, and 650 size quadrotors and set PWM_OUT in rc.mc_defaults.

### PR DESCRIPTION
**Pull request by mcsauder**
**Describe problem solved by the proposed pull request**
This PR adds a range of generic quadrotor airframes to advance PR #9849.

**Additional context**
See PR #9849 and PR #12506.

@dagar , would you let me know if you'd like to see anything different in this PR or if it is no longer of interest?  Thanks!

### CODE REVIEW

**Comment by dagar** [comment]
From a user perspective I like this idea in theory, but I think the question is if the tuning is actually going to be valid across all the other variables (battery size, motors, ESCs, props, etc) for the same frame. 

With that in mind maybe some vaguely defined generic small, medium, large sizes as a rough starting point, but not get carried away?

@MaEtUgR @bresch what do you think?

**Comment by mcsauder** [comment]
I've reduced the number of generic quad airframes previously added down to 250, 450, and 650 and moved the contents of `4050_generic_250` to `4250_generic_quad_250` (sans those values that are defaults).  Last, I've moved `set PWM_OUT 1234` to the `rc.mc_default` file, removed it from all of the other quadrotor airframe files that call `rc.mc_defaults`, and attended to `4013_bebop` which was the single exception requiring `set PWM_OUT none`.

Let me know if you have any other suggestions I could attend to with this PR.  Thanks!

**Comment by mcsauder** [comment]
@dagar, @MaEtUgR , and @bresch , it fits in flash for the moment... ;)

**Comment by mcsauder** [comment]
@dagar, if we don't go ahead with this one, what do you think about closing out this PR and #9849?

File: ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# @type Quadrotor x
+# @class Copter
+#
+
+sh /etc/init.d/rc.mc_defaults
+
+if [ $AUTOCNF = yes ]
+then
+	param set MC_ROLL_P 6.5
+	param set MC_ROLLRATE_P 0.05
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650]: That is exremely low. Did you test this?

File: ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	param set MC_ROLLRATE_I 0.05
+	param set MC_ROLLRATE_D 0.001
+
+	param set MC_PITCH_P 6.5
+	param set MC_PITCHRATE_P 0.05
+	param set MC_PITCHRATE_I 0.05
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/4650_generic_quad_650]: Please keep all I gains default (only where set higher than the default (0.2), you can keep it)

File: ROMFS/px4fmu_common/init.d/airframes/4450_generic_quad_450 [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# @type Quadrotor x
+# @class Copter
+#
+
+sh /etc/init.d/rc.mc_defaults
+
+if [ $AUTOCNF = yes ]
+then
+	param set MC_ROLL_P 7.0
+	param set MC_ROLLRATE_P 0.15
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/4450_generic_quad_450]: These are all default, can you drop them?

File: ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250 [hunk 4]
```diff
@@ -35,18 +35,17 @@ then
...
-	param set SDLOG_PROFILE 19
-
 	# disable RC filtering
 	param set RC_FLT_CUTOFF 0
 
-	param set CBRK_IO_SAFETY 22027
+	# enable high-rate logging profile (helps with tuning)
+	param set SDLOG_PROFILE 19
+
+	# use thrust curve factor (instead of TPA)
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250]:
```suggestion
	# use thrust curve factor
```

File: ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250 [hunk 5]
```diff
@@ -1,32 +1,32 @@
 #!/bin/sh
 #
-# @name Generic 250 Racer
+# @name Generic Quadcopter (250mm)
```
Comment (bkueng) [inline ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250]: This is specifically for racers, can you keep it in the name?

Comment (mcsauder) [reply ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250]: Done. Thanks!

File: ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250 (hunk 4 above)
Comment (mcsauder) [reply ROMFS/px4fmu_common/init.d/airframes/4250_generic_quad_250]: Done. Thanks!

File: ROMFS/px4fmu_common/init.d/airframes/4450_generic_quad_450 (hunk 3 above)
Comment (mcsauder) [reply ROMFS/px4fmu_common/init.d/airframes/4450_generic_quad_450]: I think @dagar 's original intent here was basically a placeholder.  I don't have a 450 airframe to know if the default values are appropriate.  Do we know of any other good candidate 450 sized airframes aside from the flame wheel that might need different parameter values?

For now, I'll add a commit that drops these values and also drops the default values from `4011_dji_f450`.  Thanks @bkueng !

File: ROMFS/px4fmu_common/init.d/airframes/4011_dji_f450 [hunk 6]
```diff
@@ -10,24 +10,10 @@
...
-	param set MC_ROLLRATE_D 0.003
-	param set MC_PITCH_P 7
-	param set MC_PITCHRATE_P 0.15
-	param set MC_PITCHRATE_I 0.05
-	param set MC_PITCHRATE_D 0.003
-	param set MC_YAW_P 2.8
-	param set MC_YAWRATE_P 0.3
-	param set MC_YAWRATE_I 0.1
-	param set MC_YAWRATE_D 0
-
```
Comment (mcsauder) [inline ROMFS/px4fmu_common/init.d/airframes/4011_dji_f450]: For explanation of these deletions, all of these are all default parameter values.

**Comment by mcsauder** [comment]
Closing this; it can be revisited in the future.  Thanks for your reviews!

**Pull request closed without merging**